test/join.test.ts

```diff
@@ -5,38 +5,35 @@ describe("joinURL", () => {
   const tests = [
     { input: [], out: "" },
     { input: ["/"], out: "/" },
-    // eslint-disable-next-line unicorn/no-null
-    { input: [null, "./"], out: "./" },
+    { input: [undefined, "./"], out: "./" },
     { input: ["./", "a"], out: "./a" },
     { input: ["./a", "./b"], out: "./a/b" },
     { input: ["/a"], out: "/a" },
     { input: ["a", "b"], out: "a/b" },
     { input: ["/", "/b"], out: "/b" },
-    { input: ["/a", "../b"], out: "/b" },
-    { input: ["../a", "../b"], out: "../b" },
-    { input: ["../a", "./../b"], out: "../b" },
-    { input: ["../a", "./../../b"], out: "b" },
-    { input: ["../a", "../../../b"], out: "../b" },
-    { input: ["../a", "../../../../b"], out: "../../b" },
-    { input: ["../a/", "../b"], out: "../b" },
-    { input: ["/a/b/c", "../../d"], out: "/a/d" },
-    { input: ["/c", "../../d"], out: "../d" },
-    { input: ["/c", ".././../d"], out: "../d" },
     { input: ["a", "b/", "c"], out: "a/b/c" },
     { input: ["a", "b/", "/c"], out: "a/b/c" },
     { input: ["/", "./"], out: "/" },
     { input: ["/", "./foo"], out: "/foo" },
     { input: ["/", "./foo/"], out: "/foo/" },
     { input: ["/", "./foo", "bar"], out: "/foo/bar" },
+
+    // Relative with ../
+    { input: ["/a", "../b"], out: "/b" },
+    { input: ["/a/b/c", "../../d"], out: "/a/d" },
+    { input: ["/c", "../../d"], out: "/d" },
```

Comment:
Updated behavior from #217. 

Also confirming two references:

```js
new URL("../../d", 'http://test.com/c').pathname
// ~> "/d"

// path.join("/c", "../../d")
// ~> "/d"
```

It is hard to decide if we should make a new 3rd behavior with `ufo` (which is really hard to break later)

Reply:
I think existing behaviour is not to coerce a URL segment to be absolute, so `joinURL('./a', 'b')` resolves to `./a/b`. Likewise, I think `joinURL('../a', 'b')` should also resolve to `../a/b` and `joinURL('../a', '../b')` should resolve to `../b`.

The key difference between `new URL` and `joinURL` is that `joinURL` is useable for URL _segments_ and this is quite a useful feature. It's also important when generating relative URLs, which is something that the native `URL` can't do.

I would advise keeping this behaviour.